src/extra/AutoUI/Lenses/types/index.ts

```diff
@@ -1,2 +1,26 @@
+import { TableColumn } from '../../../../components/Table';
+import { AutoUIContext, AutoUIModel } from '../../schemaOps';
 export { table } from './table';
 export { entity } from './entity';
+
+export interface CollectionLensRendererProps<T> {
+	filtered: T[];
+	selected: T[];
+	properties: Array<TableColumn<T>>;
+	hasUpdateActions: boolean;
+	changeSelected: (selected: T[]) => void;
+	data?: T[];
+	autouiContext: AutoUIContext<T>;
+	onEntityClick: (
+		entity: T,
+		event: React.MouseEvent<HTMLAnchorElement, MouseEvent>,
+	) => void;
+}
+
+export interface EntityLensRendererProps<T> {
+	data: T;
+	properties: Array<any & { key: string }>;
+	autouiContext: AutoUIContext<T>;
+	model: AutoUIModel<T>;
+	hasUpdateActions: boolean;
+}
```

Comment:
Most of these props are the same typing-wise. The only differences are `data` and `properties`. If you want we can have a `BaseLensRendererProps` and make CollectionLens and EntityLens extend it with their own variations of `data` and `properties`.

I also would not use the name `CollectionLens` because collection refers to the data being passed, not the way it is being rendered. The lens is a Table lens because it renders a table (using a collection of data). The Entity lens was an iffy one and I am not a fan of keeping it's name as Entity, I just was not sure what else to name it at the time. Perhaps Summary lens would be better?

What I am trying to get at with this comment and my other comment above is that each lens should be named appropriately as per what it renders and it should accept props as necessary rather than being restricted to these two interfaces which were made specifically for the `table` and `entity` lenses. For example, see the two custom lenses we currently have in the UI: Grid lens for ApplicationList.tsx and Map lens for Devices.tsx. The props their renderers accept differ from the default `table` lens props. The Grid lens only needs `filtered`, nothing else. And the Map lens only needs `filtered` and `autouicontext`.

Reply:
> I also would not use the name CollectionLens because collection refers to the data being passed, not the way it is being rendered. The lens is a Table lens because it renders a table (using a collection of data). 

This was exactly why the interface should not be called TableLense. B/c uses can define their own lense which might render a different result. A different way to think this is that this is the interface that all lenses that are supposed to render a collection can use, regardless of whether its a table or whether it's a ApplicationBox DataGrid. The default TableLense is just the backed in default lense that we happen to offer as part of AutoUI.

Now for the UI custom lense that defines only `filtered` in its prop typings, that only works b/c the typings atm for the props are just `any`.